### PR TITLE
CI update: Adding composer cache and refactoring front sniffers

### PR DESCRIPTION
**Pull request by davidferlay**
TLDR :
- **Time required for `prepare` + `deploy` CI steps was reduced from 4 min 56 sec to 2 min 34 sec**

Details : 
- `prepare:back` was added to leverage cache, just like `prepare:front`
   - `deploy:review` new make uses of artefacts from both `prepare:back` and `prepare:back` jobs prepared in previous step
- `make composer` Makefile target was renamed `make back` for homogenization
- `sniffers:front` CI job was reworked to not run directly in node image
   - to benefit from proper gitlab-ci caching (as container based cache volumes creation is disabled by default)
   - to fix [artifacts ownership](https://gitlab.com/gitlab-org/gitlab-runner/issues/1180) issue from one job to another
- Arbitrary string cache keys were replaced by file based (SHA checksum tags) cache keys
   - to create more meaningful cache invalidation and share cached dependencies between all branch of projects AMAP
- `make front` was splitted into `make front-install` and `make front-build` in order to execute only `make front-build` in prepare:front CI step and save additional resources
- `job_deploy_commit` anchor template was removed from `deploy:review` since
   - RA lifecycle is managed by bot in our case and doesn't seem necessary according to tests
   - it's useful content was moved to `stop_review`
- `artifacts` archived were named differently for back and front dependencies using predefined gitlab variables
- Faulty `make all MAKE_ENV=ci` command was removed to be fixed in another PR
- Missing _.PHONY_ commands were added (mostly from front.mk)
- `pwd` and `ls -lah` debug commands were cleaned for style

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -96,38 +98,63 @@ sniffers:sonarqube:
...
+    - docker-compose --version
+    - echo "INSTALL_DEV_DEPENDENCIES=TRUE" >> .env.default
+    - make back
+    - docker-compose down # PHP container was required for composer
+  cache:
+    key:
+      files:
+        - composer.json
+        - composer.lock
+    paths:
```
Comment (piggito) [inline .gitlab-ci.yml]: I believe cache paths should be the same as artifact paths cause remember that scaffold creates files on web/ folder root level

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -96,38 +98,63 @@ sniffers:sonarqube:
...
+        - ${THEME_PATH}/yarn.lock
+    paths:
+      - ${THEME_PATH}/node_modules/ # Populated during yarn install
   dependencies:
-    - sniffers:phpcs
+    - sniffers:front
+  artifacts:
+    name: "$CI_COMMIT_REF_NAME:$CI_COMMIT_SHA:front"
+    paths:
+      - ${THEME_PATH}/node_modules/ # Populated during yarn install
```
Comment (piggito) [inline .gitlab-ci.yml]: Do we really need to add node packages to artifacts? After this point we don't seem to need them anymore cause lint and build already passed.

Comment (iberdinsky-skilld) [reply .gitlab-ci.yml]: sliderInit

Comment (davidferlay) [reply .gitlab-ci.yml]: Makes total sense ! After assets get compiled by `yarn build` > `node_modules` dir is not useful for RA, only `dist` dir is.
Gonna make a follow up PR
